### PR TITLE
[gdb coordinator] Fix race condition between console and OpenOCD

### PR DESCRIPTION
**Pull request by pamaury**
A previous commit introduce a one second delay between the start of the console, and the start of OpenOCD, to prevent interference when opentitantool reconfigures GPIOs as part of the init when starting the console. This commit makes this fix more robust by replacing the fixed delay with a proper sync. We wait until opentitantool prints a specific message indicating that it is ready. This message occurs after the GPIO reconfiguration which ensures proper synchronization.

### CODE REVIEW

**Comment by pamaury** [comment]
This should close #18087

**Review by marnovandermaas — APPROVED** [review]
LGTM, it would be good if someone can test this. Could you decrease the length of your commit message to fit on one line (for example by removing "properly")?

**Review by timothytrippel — APPROVED** [review]

File: rules/scripts/gdb_test_coordinator.py [hunk 1]
```diff
@@ -205,8 +205,14 @@ def gdb_maybe_consume_expected_line(line: str):
     # with the JTAG tap because opentitantool will reconfigure some GPIOs and
     # this will interfere with the JTAG tap (#17729)
     background = BackgroundProcessGroup()
-    background.run(console_command, "CONSOLE", COLOR_RED)
-    time.sleep(1)
+    console = background.run(console_command, "CONSOLE", COLOR_RED)
+    # Wait until the console is ready to display message (at this point we know
+    # it won't touch the GPIOs anymore)
+    console_ready = background.block_until_line_contains(
```
Comment (dmcardle) [inline rules/scripts/gdb_test_coordinator.py:211]: Nit: Could you move the comment that starts with "Run OpenOCD, GDB, and the OpenTitanTool console in the background." above line 208?

**Review by dmcardle — APPROVED** [review]
LGTM w/ nit

**Comment by dmcardle** [comment]
> LGTM, it would be good if someone can test this.

@marnovandermaas This should be exercised by CI's "Experimental CW310 ROM JTAG Tests". But we need to explicitly check the results because they are non-fatal to the CI run.

**Comment by timothytrippel** [comment]
Just FYI, once this goes in, I will be making the JTAG CI tests blocking.

File: rules/scripts/gdb_test_coordinator.py (hunk 1 above)
Comment (pamaury) [reply rules/scripts/gdb_test_coordinator.py:211]: Will do.

**Comment by pamaury** [comment]
I will let the CI finish its run before I push an update that will fix the commit line length and the comment in the file.

**Pull request merged**